items:
- uid: System.DllNotFoundException
  commentId: T:System.DllNotFoundException
  id: DllNotFoundException
  parent: System
  children:
  - System.DllNotFoundException.#ctor
  - System.DllNotFoundException.#ctor(System.String)
  - System.DllNotFoundException.#ctor(System.String,System.Exception)
  langs:
  - csharp
  - vb
  name: DllNotFoundException
  fullName: System.DllNotFoundException
  type: Class
  source:
    remote:
      path: src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: DllNotFoundException
    path: src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs
    startLine: 19
  assemblies:
  - System.Runtime.InteropServices
  namespace: System
  summary: "\n<p>The exception that is thrown when a DLL specified in a DLL import cannot be found.</p>\n"
  remarks: "\n<p>\n<xref href=\"System.DllNotFoundException\" data-throw-if-not-resolved=\"false\"></xref> uses the HRESULT COR_E_DLLNOTFOUND, which has the value 0x80131524.</p>\n<p>For a list of initial property values for an instance of <xref href=\"System.DllNotFoundException\" data-throw-if-not-resolved=\"false\"></xref>, see the <xref href=\"System.DllNotFoundException.#ctor\" data-throw-if-not-resolved=\"false\"></xref> constructors.</p>\n<p>The <xref href=\"System.DllNotFoundException\" data-throw-if-not-resolved=\"false\"></xref> is thrown when a dynamic link library defined by the <xref href=\"System.Runtime.InteropServices.DllImportAttribute\" data-throw-if-not-resolved=\"false\"></xref> attribute or a comparable language construct, such as the Visual Basic <strong>Declare</strong> statement, cannot be found. It is not thrown when the common language runtime is unable to load a .NET Framework assembly.</p>\n<p>Note that the <xref href=\"System.Runtime.InteropServices.DllImportAttribute\" data-throw-if-not-resolved=\"false\"></xref> can specify the path to the directory in which the DLL resides. If a path is not specified, Windows uses the search order described in <see href=\"http://go.microsoft.com/fwlink/?LinkId=122531\">Dynamic-Link Library Search Order</see>. </p>\n"
  syntax:
    content: 'public class DllNotFoundException : TypeLoadException'
    content.vb: >-
      Public Class DllNotFoundException
          Inherits TypeLoadException
  seealso:
  - type: System.Exception
    commentId: T:System.Exception
  inheritance:
  - System.Object
  modifiers.csharp:
  - public
  - class
  modifiers.vb:
  - Public
  - Class
- uid: System.DllNotFoundException.#ctor
  commentId: M:System.DllNotFoundException.#ctor
  id: '#ctor'
  parent: System.DllNotFoundException
  langs:
  - csharp
  - vb
  name: DllNotFoundException()
  fullName: System.DllNotFoundException.DllNotFoundException()
  type: Constructor
  source:
    remote:
      path: src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: .ctor
    path: src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs
    startLine: 21
  assemblies:
  - System.Runtime.InteropServices
  namespace: System
  summary: "\n<p>Initializes a new instance of the <xref href=\"System.DllNotFoundException\" data-throw-if-not-resolved=\"false\"></xref> class with default properties.</p>\n"
  remarks: "\n<p>The following table shows the initial property values for an instance of <xref href=\"System.DllNotFoundException\" data-throw-if-not-resolved=\"false\"></xref>.</p>\n<p>Property </p>\n<p>Value </p>\n<table><tbody><tr><td>\n<p>\n<xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>A null reference (<strong>Nothing</strong> in Visual Basic). </p>\n</td></tr><tr><td>\n<p>\n<xref href=\"System.TypeLoadException.Message\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>The localized error message string. </p>\n</td></tr></tbody></table>\n"
  syntax:
    content: public DllNotFoundException()
    content.vb: Public Sub New
  modifiers.csharp:
  - public
  modifiers.vb:
  - Public
- uid: System.DllNotFoundException.#ctor(System.String)
  commentId: M:System.DllNotFoundException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.DllNotFoundException
  langs:
  - csharp
  - vb
  name: DllNotFoundException(String)
  fullName: System.DllNotFoundException.DllNotFoundException(System.String)
  type: Constructor
  source:
    remote:
      path: src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: .ctor
    path: src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs
    startLine: 22
  assemblies:
  - System.Runtime.InteropServices
  namespace: System
  summary: "\n<p>Initializes a new instance of the <xref href=\"System.DllNotFoundException\" data-throw-if-not-resolved=\"false\"></xref> class with a specified error message.</p>\n"
  remarks: "\n<p>The following table shows the initial property values for an instance of <xref href=\"System.DllNotFoundException\" data-throw-if-not-resolved=\"false\"></xref>.</p>\n<p>Property </p>\n<p>Value </p>\n<table><tbody><tr><td>\n<p>\n<xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>A null reference (<strong>Nothing</strong> in Visual Basic). </p>\n</td></tr><tr><td>\n<p>\n<xref href=\"System.TypeLoadException.Message\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>The error message string. </p>\n</td></tr></tbody></table>\n"
  syntax:
    content: public DllNotFoundException(string message)
    content.vb: Public Sub New(message As String)
    parameters:
    - id: message
      type: System.String
      description: "\n<p>The error message that explains the reason for the exception. </p>\n"
  modifiers.csharp:
  - public
  modifiers.vb:
  - Public
- uid: System.DllNotFoundException.#ctor(System.String,System.Exception)
  commentId: M:System.DllNotFoundException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.DllNotFoundException
  langs:
  - csharp
  - vb
  name: DllNotFoundException(String, Exception)
  fullName: System.DllNotFoundException.DllNotFoundException(System.String, System.Exception)
  type: Constructor
  source:
    remote:
      path: src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: .ctor
    path: src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs
    startLine: 23
  assemblies:
  - System.Runtime.InteropServices
  namespace: System
  summary: "\n<p>Initializes a new instance of the <xref href=\"System.DllNotFoundException\" data-throw-if-not-resolved=\"false\"></xref> class with a specified error message and a reference to the inner exception that is the cause of this exception.</p>\n"
  remarks: "\n<p>An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref> property. The <xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref> property returns the same value that is passed into the constructor, or <strong>null</strong> if the <xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref> property does not supply the inner exception value to the constructor.</p>\n<p>The following table shows the initial property values for an instance of <xref href=\"System.DllNotFoundException\" data-throw-if-not-resolved=\"false\"></xref>.</p>\n<p>Property </p>\n<p>Value </p>\n<table><tbody><tr><td>\n<p>\n<xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>The inner exception reference. </p>\n</td></tr><tr><td>\n<p>\n<xref href=\"System.TypeLoadException.Message\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>The error message string. </p>\n</td></tr></tbody></table>\n"
  syntax:
    content: public DllNotFoundException(string message, Exception inner)
    content.vb: Public Sub New(message As String, inner As Exception)
    parameters:
    - id: message
      type: System.String
      description: "\n<p>The error message that explains the reason for the exception. </p>\n"
    - id: inner
      type: System.Exception
      description: "\n<p>The exception that is the cause of the current exception. If the <em>inner</em> parameter is not <strong>null</strong>, the current exception is raised in a <strong>catch</strong> block that handles the inner exception. </p>\n"
  seealso:
  - type: System.Exception
    commentId: T:System.Exception
  modifiers.csharp:
  - public
  modifiers.vb:
  - Public
references:
- uid: System
  isExternal: false
  name: System
  fullName: System
- uid: System.Object
  parent: System
  isExternal: false
  name: Object
  fullName: System.Object
- uid: System.String
  parent: System
  isExternal: false
  name: String
  fullName: System.String
- uid: System.Exception
  parent: System
  isExternal: false
  name: Exception
  fullName: System.Exception
